hashes/src/hkdf/mod.rs

```diff
@@ -106,6 +106,26 @@ where
     }
 }
 
+impl<T: GeneralHash> fmt::Debug for Hkdf<T> {
+    fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+        use crate::{sha256t, sha256t_tag};
+
+        struct Fingerprint([u8; 8]); // Print 16 hex characters as a fingerprint.
+
+        impl fmt::Debug for Fingerprint {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { crate::debug_hex(&self.0, f) }
+        }
+
+        sha256t_tag! {
+            pub struct Tag = hash_str("bitcoin_hashes1DEBUG");
+        }
+
+        let hash = sha256t::Hash::<Tag>::hash(self.prk.as_ref());
+        let fingerprint = Fingerprint(core::array::from_fn(|i| hash.as_byte_array()[i]));
```

Comment:
Honestly, I find this confusing. It looks like a copy, which made me think "WTF, why not just `to_byte_array`?" while semantically it's subslicing - could be simply `hash.as_byte_array()[..8].try_into().unwrap()` to show the slicing operation in the code.

Reply:
Yeah, sorry, I considered this and then was tempted to be clever by avoiding the `mut`, the 0-initialization *and* the explicit panic branch.